python/plugins/processing/algs/qgis/HubDistanceLines.py

```diff
@@ -103,6 +103,10 @@ def processAlgorithm(self, parameters, context, feedback):
         hub_source = self.parameterAsSource(parameters, self.HUBS, context)
         if hub_source is None:
             raise QgsProcessingException(self.invalidSourceError(parameters, self.HUBS))
+        if hub_source.featureCount() <= 0:
```

Comment:
A value of -1 for featureCount indicates "not known", not "the layer is empty". Could you use hub_source.hasFeatures() instead here?

Reply:
Thanks for your hint!

I've replaced `hub_source.featureCount() <= 0` with `hub_source.hasFeatures() == QgsFeatureSource.FeatureAvailability.NoFeaturesAvailable`

Is it possible that `hub_source.hasFeatures() == QgsFeatureSource.FeatureAvailability.FeaturesMaybeAvailable` ([reference](https://qgis.org/pyqgis/3.22/core/QgsFeatureSource.html#qgis.core.QgsFeatureSource.FeaturesMaybeAvailable)) but there aren't any features available? Because then the code might fail again in line 139.